r2-core/src/main/java/com/linkedin/r2/transport/http/client/ConstantQpsRateLimiter.java

```diff
@@ -97,5 +100,10 @@ public int getMaxBuffered()
     {
       return Integer.MAX_VALUE;
     }
+
+    public int getNextExecutionDelay(Rate rate)
```

Comment:
This seems to me only adds a delay before the first execution. The naming seems confusing here.  Maybe `getInitialDelay()`?

Reply:
The intent of this method is indeed to provide a delay in ms to be incurred before every execution.

I'd like to get your opinion here. Given that this method requires intimate knowledge of a Rate, it may make more sense to make it a static method in Rate called "getRandomAcceptableDelay" or something like that. Basically something that clearly describes this delay as a random delay no greater than the amount of free time in a given period while still satisfying the rate.

A degree of randomness is ideal in any traffic synthetically generated, and it's possible that there may be synthetic use cases down the line that aren't handled by ConstantQpsRateLimiter.

If we did go this route, maybe we could then convert RateLimiterExecutionTracker.getNextExecutionDelay into a boolean getter that is used to decide whether or not we call the new static method in Rate.

Reply:
I would prefer to keep `Rate` as is. It's more like a configuration class to me